gson/src/main/java/com/google/gson/ToNumberPolicy.java

```diff
@@ -68,22 +68,30 @@ public Number readNumber(JsonReader in) throws IOException {
     @Override
     public Number readNumber(JsonReader in) throws IOException, JsonParseException {
       String value = in.nextString();
-      try {
-        return Long.parseLong(value);
-      } catch (NumberFormatException longE) {
+      if (value.indexOf('.') >= 0) {
```

Comment:
I think this makes sense. We could also imagine a more thorough check or even a regular expression like `[-+]?[0-9]+`, but I think looking for just `'.'` is an excellent choice.

Reply:
Agreed. Since I'm looking for performance, the search for `'.'` is faster than applying a regexp. The `Double.valueOf()` will take care of anything else, but we only call it if there are chances of being a double